libs/gi/page-weapons/src/WeaponFilter.tsx

```diff
@@ -0,0 +1,335 @@
+                        <Trans t={t} i18nKey={`ui:${v}`} />
+                        <Chip
+                          label={lockedTotal[i ? 'unlocked' : 'locked']}
+                          size="small"
+                        />
+                      </ToggleButton>
+                    ))}
+                  </SolidToggleButtonGroup>
+                  <Button
+                    startIcon={<BusinessCenterIcon />}
+                    color={showInventory ? 'success' : 'secondary'}
+                    onClick={() =>
+                      database.displayWeapon.set({
+                        showInventory: !showInventory,
+                      })
+                    }
+                  >
+                    {t`weaponInInv`}{' '}
+                    <Chip
+                      sx={{ ml: 1 }}
+                      label={equippedTotal['unequipped']}
+                      size="small"
+                    />
+                  </Button>
+                  <Button
+                    startIcon={<PersonSearchIcon />}
+                    color={showEquipped ? 'success' : 'secondary'}
+                    onClick={() =>
+                      database.displayWeapon.set({
+                        showEquipped: !showEquipped,
+                      })
+                    }
+                  >
+                    {t`equippedWeapon`}{' '}
+                    <Chip
+                      sx={{ ml: 1 }}
+                      label={equippedTotal['equipped']}
+                      size="small"
+                    />
+                  </Button>
+                </Stack>
+              </Box>
+            </Grid>
+          </Grid>
+        </Box>
+      </CardContent>
+    </CardThemed>
+  )
+}
+
+function WeaponRedButtons({ weaponIds }: { weaponIds: string[] }) {
+  const { t } = useTranslation(['page_weapon', 'ui'])
+  const database = useDatabase()
+  const { numDelete, numUnlock, numLock } = useMemo(() => {
+    const weapons = weaponIds.map((id) =>
+      database.weapons.get(id)
+    ) as ICachedWeapon[]
+    const numUnlock = weapons.reduce(
+      (w, weapon) => w + (weapon.lock ? 0 : 1),
+      0
+    )
+    const numLock = weapons.length - numUnlock
+    const numDelete = weapons.reduce(
+      (w, weapon) => w + (weapon.lock || weapon.location ? 0 : 1),
+      0
+    )
+    return { numDelete, numUnlock, numLock }
+  }, [weaponIds, database])
+
+  const deleteWeapons = () =>
+    window.confirm(`Are you sure you want to delete ${numDelete} weapons?`) &&
+    weaponIds.map((id) => {
+      const weapon = database.weapons.get(id)
+      if (!weapon?.lock && !weapon?.location) database.weapons.remove(id)
+    })
+
+  const lockWeapons = () =>
+    window.confirm(`Are you sure you want to lock ${numLock} weapons ?`) &&
+    weaponIds.map((id) => database.weapons.set(id, { lock: true }))
+
+  const unlockWeapons = () =>
+    window.confirm(`Are you sure you want to unlock ${numLock} weapons ?`) &&
```

Comment:
![image](https://github.com/frzyc/genshin-optimizer/assets/1754901/2c159077-2276-45d9-bf1e-d45824bfb07d)
The comfirm prompt shows 0, but its 265 weapons to be locked

Reply:
Fixed